hack/generator/pkg/astmodel/armconversion/convert_to_arm_function_builder.go

```diff
@@ -0,0 +1,556 @@
+				return true, nil
+			}
+
+			return false, nil
+		},
+		matchHandler: func(toProp *astmodel.PropertyDefinition, fromProp *astmodel.PropertyDefinition) []ast.Stmt {
+			propertyType := toProp.PropertyType()
+			if optionalType, ok := toProp.PropertyType().(*astmodel.OptionalType); ok {
+				propertyType = optionalType.Element()
+			}
+
+			enumTypeName, ok := propertyType.(astmodel.TypeName)
+			if !ok {
+				panic(fmt.Sprintf("Type property was not an enum, was %T", toProp.PropertyName()))
+			}
+
+			def, err := builder.codeGenerationContext.GetImportedDefinition(enumTypeName)
+			if err != nil {
+				panic(err)
+			}
+
+			enumType, ok := def.Type().(*astmodel.EnumType)
+			if !ok {
+				panic(fmt.Sprintf("Enum %v definition was not of type EnumDefinition", enumTypeName))
+			}
+
+			optionId := astmodel.GetEnumValueId(def.Name(), enumType.Options()[0])
+
+			result := astbuilder.SimpleAssignment(
+				&ast.SelectorExpr{
+					X:   builder.resultIdent,
+					Sel: ast.NewIdent(string(toProp.PropertyName())),
+				},
+				token.ASSIGN,
+				ast.NewIdent(optionId))
+
+			return []ast.Stmt{result}
+		},
+	}
+}
+
+func (builder *convertToArmBuilder) propertiesWithSameNameAndTypeHandler() propertyConversionHandler {
+
+	return propertyConversionHandler{
+		finder: func(toProp *astmodel.PropertyDefinition, fromType *astmodel.ObjectType) (bool, *astmodel.PropertyDefinition) {
+			fromProp, ok := fromType.Property(toProp.PropertyName())
+			if ok && toProp.PropertyType().Equals(fromProp.PropertyType()) {
+				return true, fromProp
+			}
+			return false, nil
+		},
+		matchHandler: func(toProp *astmodel.PropertyDefinition, fromProp *astmodel.PropertyDefinition) []ast.Stmt {
+			result := astbuilder.SimpleAssignment(
+				&ast.SelectorExpr{
+					X:   builder.resultIdent,
+					Sel: ast.NewIdent(string(toProp.PropertyName())),
+				},
+				token.ASSIGN,
+				&ast.SelectorExpr{
+					X:   builder.receiverIdent,
+					Sel: ast.NewIdent(string(fromProp.PropertyName())),
+				})
+
+			return []ast.Stmt{result}
+		},
+	}
+}
+
+func (builder *convertToArmBuilder) propertiesWithSameNameButDifferentTypeHandler() propertyConversionHandler {
+
+	return propertyConversionHandler{
+		finder: func(toProp *astmodel.PropertyDefinition, fromType *astmodel.ObjectType) (bool, *astmodel.PropertyDefinition) {
+			fromProp, ok := fromType.Property(toProp.PropertyName())
+			if ok && !toProp.PropertyType().Equals(fromProp.PropertyType()) {
+				return true, fromProp
+			}
+			return false, nil
+		},
+		matchHandler: func(toProp *astmodel.PropertyDefinition, fromProp *astmodel.PropertyDefinition) []ast.Stmt {
+			destination := &ast.SelectorExpr{
+				X:   builder.resultIdent,
+				Sel: ast.NewIdent(string(toProp.PropertyName())),
+			}
+			source := &ast.SelectorExpr{
+				X:   builder.receiverIdent,
+				Sel: ast.NewIdent(string(fromProp.PropertyName())),
+			}
+
+			return builder.toArmComplexPropertyConversion(
```

Comment:
You've got it - `:=` is always safe in one context because at least one variable is guaranteed to be new. In the other context (`FromArm`) it's not because we're not declaring any new vars.